examples/custom_code_deployment/README.md

```diff
@@ -49,28 +68,548 @@ in the integration example.
+described in this document, but similar stacks may be set up using different 
+backends and used to run the example as long as the basic Stack prerequisites 
+are met.
+
+This stack consists of the following components:
+
+* a GCP artifact store
+* the local orchestrator
+* the local metadata store
+* a KServe model deployer
+* a local secret manager used to store the credentials needed by KServe to
+access the GCP artifact store
+* a GCP container registry used to store the custom docker images used by the
+KServe model deployer
+
+To have access to the GCP artifact store from your local workstation, the
+gcloud client needs to be properly set up locally.
+
+In addition to the stack components, KServe must be installed in a
+Kubernetes cluster that is locally accessible through a Kubernetes configuration
+context. The reference used in this example is a KServe installation
+running in a GKE cluster, but any other type of Kubernetes cluster can be used,
+managed or otherwise.
+
+To configure GKE cluster access locally, e.g:
+
+```bash
+gcloud container clusters get-credentials zenml-test-cluster --zone us-east1-b --project zenml-core
+```
+
+Set up a namespace for ZenML KServe workloads:
+
+```bash
+kubectl create ns zenml-workloads
+```
+
+Extract the URL where the KServe model server exposes its prediction API, e.g.:
+
+```bash
+# If you are running in GKE or AKS clusters, the host is the GKE cluster IP address.
+export INGRESS_HOST=$(kubectl -n istio-system get service istio-ingressgateway -o jsonpath='{.status.loadBalancer.ingress[0].ip}')
+# If you are running in EKS clusters, the host is the EKS cluster IP hostname.
+export INGRESS_HOST=$(kubectl -n istio-system get service istio-ingressgateway -o jsonpath='{.status.loadBalancer.ingress[0].hostname}')
+
+export INGRESS_PORT=$(kubectl -n istio-system get service istio-ingressgateway -o jsonpath='{.spec.ports[?(@.name=="http2")].port}')
+export INGRESS_URL="http://${INGRESS_HOST}:${INGRESS_PORT}"
+```
+
+Configuring the stack can be done like this:
+
+```shell
+zenml model-deployer register kserve_gke --flavor=kserve \
+  --kubernetes_context=gke_zenml-core_us-east1-b_zenml-test-cluster \ 
+  --kubernetes_namespace=zenml-workloads \
+  --base_url=$INGRESS_URL \
+  --secret=kserve_secret
+zenml artifact-store register gcp_artifact_store --flavor=fcp --path gs://my-bucket
+zenml secrets-manager register local --flavor=local
+zenml container-registry register gcp_registry --flavor=gcp --uri=eu.gcr.io/container-registry
+zenml stack register local_gcp_kserve_stack -m default -a gcp_artifact_store -o default -d kserve_gke -c gcp_registry -x local --set
+```
+
+The next sections cover how to set GCP Artifact Store credentials for the KServe model deployer,   
+Please look up the variables relevant to your use case in the
+[official KServe Storage Credentials](https://kserve.github.io/website/0.8/sdk_docs/docs/KServeClient/#parameters)
+and set them accordingly for your ZenML secret.
```

Comment:
Does this need any more explanation or perhaps an example? Feels like maybe there is some ambiguity in here?